sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/src/main/java/com/azure/messaging/eventgrid/cloudnative/cloudevents/EventGridCloudNativeEventPublisher.java

```diff
@@ -0,0 +1,181 @@
+     * @param event the native cloud event to publish.
+     * @return a Mono that completes when the events are sent to the service.
+     */
+    public static Mono<Void> sendEventAsync(EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient,
+        CloudEvent event) {
+        return asyncClient.sendEvent(toEventGridCloudEvent(event));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     */
+    public static void sendEvents(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events) {
+        syncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return a Mono that completes when the events are sent to the service.
+     */
+    public static Mono<Void> sendEventsAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @param context the context to use along the pipeline.
+     */
+    public static void sendEventsWithResponse(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events, Context context) {
+        syncClient.sendEventsWithResponse(toEventGridCloudEvents(events), context);
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return the response from the EventGrid service.
+     */
+    public static Mono<Response<Void>> sendEventsWithResponseAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEventsWithResponse(toEventGridCloudEvents(events));
+    }
+
+    private static Iterable<com.azure.core.models.CloudEvent> toEventGridCloudEvents(Iterable<CloudEvent> events) {
+        List<com.azure.core.models.CloudEvent> cloudEvents = new ArrayList<>();
+        for (CloudEvent event : events) {
```

Comment:
This can potentially throw NPE if `events` is `null`. This should be null-checked in the API above and include an error message.